Frontend/src/Login.jsx

```diff
@@ -20,19 +20,35 @@ export default function Login({setLoginTime, apiKey, setApiKey, setUserData}) {
     // alert("Username: " + username + " " + "Password: " + password);
   };
 
+  useEffect(() => {
+    const login = async () => {
+      const temp = await axios.post(`${API_URL}/loginUser`, {
+        username: username,
+        password: password,
+        apiKey: apiKey
+      })
+  
+      console.log(temp)
+      
+      setUserData(temp.data.userData)  
+      setUserId(temp.data.userId)
+      setCourses(temp.data.courses)
+    }
 
-  const attemptLogin = async (method) => {
-    setLoginTime(Date.now());
+    if (apiKey && logState == "login") {
+      login()
+    }
+
+  }, [apiKey, logState])
```

Comment:
This still has a problem if there's something in the API key box but the user cancels the registration and goes back to the login page.

Does this work? Seems to work when I tried it.

```diff
diff --git a/Frontend/src/Login.jsx b/Frontend/src/Login.jsx
index b2195b7..25c8544 100644
--- a/Frontend/src/Login.jsx
+++ b/Frontend/src/Login.jsx
@@ -20,35 +20,24 @@ export default function Login({apiKey, setApiKey, setUserData, setUserId, setCou
     // alert("Username: " + username + " " + "Password: " + password);
   };
 
-  useEffect(() => {
-    const login = async () => {
-      const temp = await axios.post(`${API_URL}/loginUser`, {
-        username: username,
-        password: password,
-        apiKey: apiKey
-      })
-  
-      console.log(temp)
-      
-      setUserData(temp.data.userData)  
-      setUserId(temp.data.userId)
-      setCourses(temp.data.courses)
-    }
-
-    if (apiKey && logState == "login") {
-      login()
-    }
-
-  }, [apiKey, logState])
-
 
   const attemptLogin = async (method) => {
     if (method === "login") {
       try {
-        setApiKey(localStorage.getItem("canvasAPIKey"))
-        console.log("logged in")
+        const currentKey = localStorage.getItem("canvasAPIKey");
+        setApiKey(currentKey);
+        const result = await axios.post(`${API_URL}/loginUser`, {
+          username: username,
+          password: password,
+          apiKey: currentKey
+        });
 
+        console.log(result);
         
+        setUserData(result.data.userData);
+        setUserId(result.data.userId);
+        setCourses(result.data.courses);
+        console.log("logged in");
       } catch (e) {
         console.log("login failed");
       }

```